File: ivory/src/main/java/io/happylrd/ivory/pojo/Student.java
package io.happylrd.ivory.pojo;

import javax.persistence.*;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Entity
public class Student {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @Column(unique = true, length = 30, nullable = false, updatable = false)
    private String username;

    @Column(length = 128, nullable = false)
    private String password;

    @Column(length = 10)
    private String realname;

    @Column(length = 11)
    private String mobile;

    @Column(length = 20)
    private String school;

    @Column(length = 30)
    private String major;

    /**
     * Current constraint: `male` for 男生, `female` for 女生, `unknown` for 未知.
     * will be improved later.
     */
    @Column(length = 7)
    private String gender = "unknown";

    private LocalDate birthday;

    private String avatarUrl;

    private LocalDateTime createTime;

    private LocalDateTime updateTime;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getUsername() {
        return username;
    }

    public void setUsername(String username) {
        this.username = username;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public String getRealname() {
        return realname;
    }

    public void setRealname(String realname) {
        this.realname = realname;
    }

    public String getMobile() {
        return mobile;
    }

    public void setMobile(String mobile) {
        this.mobile = mobile;
    }

    public String getSchool() {
        return school;
    }

    public void setSchool(String school) {
        this.school = school;
    }

    public String getMajor() {
        return major;
    }

    public void setMajor(String major) {
        this.major = major;
    }

    public String getGender() {
        return gender;
    }

    public void setGender(String gender) {
        this.gender = gender;
    }

    public LocalDate getBirthday() {
        return birthday;
    }

    public void setBirthday(LocalDate birthday) {
        this.birthday = birthday;
    }

    public String getAvatarUrl() {
        return avatarUrl;
    }

    public void setAvatarUrl(String avatarUrl) {
        this.avatarUrl = avatarUrl;
    }

    public LocalDateTime getCreateTime() {
        return createTime;
    }

    public void setCreateTime(LocalDateTime createTime) {
        this.createTime = createTime;
    }

    public LocalDateTime getUpdateTime() {
        return updateTime;
    }

    public void setUpdateTime(LocalDateTime updateTime) {
        this.updateTime = updateTime;
    }
}

File: ivory/src/main/kotlin/io/happylrd/ivory/pojo/Student.kt
package io.happylrd.ivory.pojo

import java.time.LocalDate
import java.time.LocalDateTime
import javax.persistence.*

@Entity
class Student(
        @Id @GeneratedValue(strategy = GenerationType.AUTO)
        var id: Long = 0,

        @Column(unique = true, length = 30, nullable = false, updatable = false)
        var username: String = "",

        @Column(length = 128, nullable = false)
        var password: String = "",

        @Column(length = 10)
        var realname: String = "",

        @Column(length = 11)
        var mobile: String = "",

        @Column(length = 20)
        var school: String = "",

        @Column(length = 30)
        var major: String = "",

        /**
         * Current constraint: `male` for 男生, `female` for 女生, `unknown` for 未知.
         * will be improved later.
         */
        @Column(length = 7)
        var gender: String = "unknown",

        var birthday: LocalDate = LocalDate.now(),

        var avatarUrl: String = "",

        var createTime: LocalDateTime = LocalDateTime.now(),

        var updateTime: LocalDateTime = LocalDateTime.now()
)